src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -244,6 +244,9 @@ public double getDistance() {
     public State traverse(State s0) {
         final RoutingRequest options = s0.getOptions();
         final TraverseMode currMode = s0.getNonTransitMode();
+        if (options.excludeWalking && currMode.equals(TraverseMode.WALK)) {
```

Comment:
We added an option to have NO walking at all, which requires the passenger to be picked up and dropped off via call-and-ride or deviated route. This option needs to be selected by a passenger, and yes, it would require that both endpoints be inside GTFS-Flex areas.

These modes are often used by mobility-impaired individuals who are unable to walk, so we had a requirement to support completely excluding walking if requested via the request parameter.